main.cc

```diff
@@ -1857,6 +1857,12 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
             stop_signal.wait().get();
             startlog.info("Signal received; shutting down");
 	    // At this point, all objects destructors and all shutdown hooks registered with defer() are executed
+          } catch (const sleep_aborted&) {
+            startlog.info("Startup interrupted");
+            // This happens when scylla gets SIGINT in the middle of join_cluster(), so
+            // just ignore it and exit normally
+            _exit(0);
+            return 0;
```

Comment:
i don't think we need to have `return 0` here. is it preserved just as a comment that 

> i could have just returned 0, but the RAII cannot help here because the ordering of destruction of main() objects and thread local objects is still not sorted in the right order.

like the other places in main() ?

Reply:
It's the dead code for sure, but I kept it to be in-sync with other places like that

Reply:
ack. will remove them all in a separated change.